test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/TagHelperComponentTagHelperTest.cs

```diff
@@ -288,5 +334,21 @@ public Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
                 return Task.CompletedTask;
             }
         }
+
+        private class TestAddTagHelperComponent : ITagHelperComponent
+        {
+            public int Order => 0;
```

Comment:
Suggest making this settable and including it in what's added to `PostContent`. Then, can test that controller-added components are processed in the correct order.